lib/core/src/Codec/Binary/Bech32/TH.hs

```diff
@@ -0,0 +1,58 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+-- |
+-- Copyright: © 2018-2020 IOHK
+-- License: Apache-2.0
+--
+-- This module contains Template-Haskell-specific extensions to the Bech32
+-- library.
+
+module Codec.Binary.Bech32.TH
+    ( humanReadablePart
+    ) where
+
+import Prelude
+
+import Codec.Binary.Bech32
+    ( HumanReadablePart, humanReadablePartFromText, humanReadablePartToText )
+import Language.Haskell.TH.Quote
+    ( QuasiQuoter (..) )
+import Language.Haskell.TH.Syntax
+    ( Lift, lift )
+
+import qualified Data.Text as T
+
+-- | A quasiquoter for Bech32 human-readable prefixes.
+--
+-- This allows the construction of prefixes to be checked at compile time.
+--
+humanReadablePart :: QuasiQuoter
+humanReadablePart = QuasiQuoter
+    { quoteExp  = lift . unsafeHumanReadablePartFromText . T.pack
+    , quotePat  = notHandled "patterns"
+    , quoteType = notHandled "types"
+    , quoteDec  = notHandled "declarations"
+    }
+  where
+    notHandled things =
+      error $ things <>
+          " are not handled by the Bech32 humanReadablePart quasiquoter."
+
+unsafeHumanReadablePartFromText :: T.Text -> HumanReadablePart
```

Comment:
can you add (or maybe recover) some testing to that? In the previous checked-at-run-time we had `unsafeHumanReadablePartFromText` in `Bench32.Internal` and there were persuasive tests in `Bech32Spec`.

Reply:
@paweljakubas yes, definitely!

IMO these tests should really be part of the testing for the `humanReadablePartFromText` function. (The `unsafeHumanReadablePartFromText` was just a thin wrapper around that.)

Since the goal here is to move away from our internal copy of the `bech32` library and instead rely on the [actual `bech32` library](https://github.com/input-output-hk/bech32), I will make a further PR against that library, adding equivalent test coverage.

Reply:
See here: https://github.com/input-output-hk/bech32/pull/7